Ingress/controllers/nginx-third-party/controller.go

```diff
@@ -0,0 +1,237 @@
+		for the backend.`)
+
+	// If you have pure tcp services or https services that need L3 routing, you
+	// must specify them by name. Note that you are responsible for:
+	// 1. Making sure there is no collision between the service ports of these services.
+	//	- You can have multiple <mysql svc name>:3306 specifications in this map, and as
+	//	  long as the service ports of your mysql service don't clash, you'll get
+	//	  loadbalancing for each one.
+	// 2. Exposing the service ports as node ports on a pod.
+	// 3. Adding firewall rules so these ports can ingress traffic.
+
+	tcpServices = flags.String("tcp-services", "", `Comma separated list of tcp/https
+		namespace/serviceName:portToExport pairings. This assumes you've opened up the right
+		hostPorts for each service that serves ingress traffic. Te value of portToExport indicates the 
+		port to listen inside nginx, not the port of the service.`)
+
+	configurationName = flags.String("configuration-name", "", `Name of the NginxIngressLoadBalancer 
+		third party resource that contains the custom configuration to be used in nginx.
+		Takes the form namespace/name.`)
+
+	healthzPort = flags.Int("healthz-port", healthPort, "port for healthz endpoint.")
+)
+
+type k8sService struct {
+	ServiceName string
+	ServicePort string
+	Namespace   string
+	ExposedPort string
+}
+
+func registerHandlers() {
+	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte("ok"))
+	})
+
+	glog.Fatalf(fmt.Sprintf("%v", http.ListenAndServe(fmt.Sprintf(":%v", *healthzPort), nil)))
+}
+
+func main() {
+	pflagGlog(flags)
+
+	flags.Parse(os.Args)
+
+	if *defaultSvc == "" {
+		glog.Fatalf("Please specify --default-backend")
+	}
+
+	var ingClient unversioned.IngressInterface
+
+	kubeClient, err := unversioned.NewInCluster()
+	if err != nil {
+		glog.Fatalf("failed to create client: %v", err)
+	} else {
+		ingClient = kubeClient.Extensions().Ingress(api.NamespaceAll)
+	}
+
+	funcMap := template.FuncMap{
+		"getSSLHost": getSSLHost,
+		"empty": func(input interface{}) bool {
+			check, ok := input.(string)
+			if ok {
+				return len(check) == 0
+			}
+
+			return true
+		},
+		"dict": func(values ...interface{}) (map[string]interface{}, error) {
+			if len(values)%2 != 0 {
+				return nil, errors.New("invalid dict call")
+			}
+			dict := make(map[string]interface{}, len(values)/2)
+			for i := 0; i < len(values); i += 2 {
+				key, ok := values[i].(string)
+				if !ok {
+					return nil, errors.New("dict keys must be strings")
+				}
+				dict[key] = values[i+1]
+			}
+			return dict, nil
+		},
+	}
+
+	tmpl, _ := template.New("nginx.tmpl").Funcs(funcMap).ParseFiles("./nginx.tmpl")
+	rateLimiter := util.NewTokenBucketRateLimiter(0.1, 1)
+
+	known := &extensions.IngressList{}
+	defCfg := defNginxCfgParams()
+	runningCfg := make(map[string]interface{})
+
+	sslCertificates := createSSLCerts(sslDirectory)
```

Comment:
These are a lot of semi-related variables out in the open. I'd suggest a type with some collection of fields, maybe something like: newTemplatedNginx() and it returns a templatedNginx{sslCertificates: , defaultBackend: }...